### PR TITLE
Replaces fft with rfft

### LINKED ISSUE #27: Rename single-character variable names to more informative names
Example `c` should be `coefficients`.
Another example is `mview` should be `matrixview`

In addition these functions need better docstrings.

### LINKED ISSUE #28: Make default FFT symmetric
We can carry fewer weights if the input is real-valued.

### PR DESCRIPTION
**Pull request by bischtob**
FFT is dumb, RFFT rulez. Saves half the memory for weights. This PR also refactors the Chebyshev transform in anticipation of it being able to work on arbitrary dimensions. This PR also refactors the tests and adds separate ChainRules.jl tests to consolidate adjoint testing.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -10,6 +10,7 @@ Distributions = "31c24e10-a181-5473-b8eb-7969acd0382f"
 FFTW = "7a1cc6ca-52ef-59f5-83cd-3a7055c09341"
 Flux = "587475ba-b771-5e3f-ad9e-33799f191a9c"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
Comment (charleskawczynski) [inline Project.toml]: Can we not add Revise?

Comment (bischtob) [reply Project.toml]: Oh hi Charlie

Comment (bischtob) [reply Project.toml]: :)
yes we can

**Pull request merged**